### PR TITLE
[extension_google_sign_in_as_googleapis_auth] Migrate to null safety

### PR DESCRIPTION
**Pull request by stuartmorgan**
Migrates the library and tests; the example is blocked by the lack of a null-safe `googleapis`. We can revisit once that's published.

Also fixes, and adds a test for, a bug where the parameter intended to allow overriding the requested scopes for debugging had a default value, so it would never use the underlying object's scopes as intended.

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Review by mvanbeusekom — APPROVED** [review]
LGTM

File: packages/google_sign_in/extension_google_sign_in_as_googleapis_auth/lib/extension_google_sign_in_as_googleapis_auth.dart [hunk 1]
```diff
@@ -15,15 +15,20 @@ import 'package:http/http.dart' as http;
 /// client that can be used with the rest of the `googleapis` libraries.
 extension GoogleApisGoogleSignInAuth on GoogleSignIn {
   /// Retrieve a `googleapis` authenticated client.
-  Future<googleapis_auth.AuthClient> authenticatedClient({
-    @visibleForTesting GoogleSignInAuthentication debugAuthentication,
-    @visibleForTesting List<String> debugScopes = const [],
```
Comment (ditman) [inline packages/google_sign_in/extension_google_sign_in_as_googleapis_auth/lib/extension_google_sign_in_as_googleapis_auth.dart:20]: This was indeed a bug! Given the amount of use this package is getting (close to none), I'm tempted of moving this out of flutter/plugins and into flutter/packages, so it doesn't interfere more with important parts of the repo moving forward :/

**Review by ditman — APPROVED** [review]
Thanks for migrating this one!

File: packages/google_sign_in/extension_google_sign_in_as_googleapis_auth/lib/extension_google_sign_in_as_googleapis_auth.dart (hunk 1 above)
Comment (stuartmorgan) [reply packages/google_sign_in/extension_google_sign_in_as_googleapis_auth/lib/extension_google_sign_in_as_googleapis_auth.dart:20]: I was wondering why it was here. Moving anything that's not actually a plugin to flutter/packages SGTM. (See also https://github.com/flutter/flutter/issues/76867)

Comment (ditman) [reply packages/google_sign_in/extension_google_sign_in_as_googleapis_auth/lib/extension_google_sign_in_as_googleapis_auth.dart:20]: Yeah, good one too. These should be moved to flutter/packages. I'll try to get it done while preserving history after I'm done with the nnbd _web stuff.

**Comment by ditman** [comment]
Published

**Pull request merged**